src/modules/firewall/src/lib/Firewall.h

```diff
@@ -19,169 +24,145 @@ class FirewallLog
 public:
     static OSCONFIG_LOG_HANDLE Get()
     {
-        return m_logFirewall;
+        return m_logHandle;
     }
     static void OpenLog()
     {
-        m_logFirewall = ::OpenLog(FIREWALL_LOGFILE, FIREWALL_ROLLEDLOGFILE);
+        m_logHandle = ::OpenLog(FIREWALL_LOGFILE, FIREWALL_ROLLEDLOGFILE);
     }
     static void CloseLog()
     {
-        ::CloseLog(&m_logFirewall);
+        ::CloseLog(&m_logHandle);
     }
 
 private:
-    static OSCONFIG_LOG_HANDLE m_logFirewall;
+    static OSCONFIG_LOG_HANDLE m_logHandle;
 };
 
-class Rule
+class GenericUtility
 {
 public:
-    Rule(int ruleNum, std::string target, std::string protocol, std::string source, std::string destination, std::string sourcePort, std::string destinationPort, std::string inInterface, std::string outInterface, std::string rawOptions);
-    Rule();
-    void SetRuleNum(int ruleNum);
-    void SetTarget(std::string target);
-    void SetProtocol(std::string protocol);
-    void SetSource(std::string source);
-    void SetDestination(std::string destination);
-    void SetSourcePort(std::string sourcePort);
-    void SetDestinationPort(std::string destinationPort);
-    void SetInInterface(std::string inInterface);
-    void SetOutInterface(std::string outInterface);
-    void SetRawOptions(std::string rawOptions);
-
-    int GetRuleNum();
-    std::string GetTarget();
-    std::string GetProtocol();
-    std::string GetSource();
-    std::string GetDestination();
-    std::string GetSourcePort();
-    std::string GetDestinationPort();
-    std::string GetInInterface();
-    std::string GetOutInterface();
-    std::string GetRawOptions();
+    enum class State
+    {
+        Unknown = 0,
+        Enabled,
+        Disabled
+    };
 
-private:
-    int m_ruleNum;
-    std::string m_policy;
-    std::string m_target;
-    std::string m_protocol;
-    std::string m_source;
-    std::string m_destination;
-    std::string m_sourcePort;
-    std::string m_destinationPort;
-    std::string m_inInterface;
-    std::string m_outInterface;
-    std::string m_rawOptions;
-};
+    virtual ~GenericUtility() = default;
 
-enum FirewallStateCode
-{
-    firewallStateCodeUnknown = 0,
-    firewallStateCodeEnabled,
-    firewallStateCodeDisabled
+    virtual State Detect() const = 0;
+    virtual std::string Hash() const = 0;
 };
 
-enum UtilityStatusCode
+class IpTables : public GenericUtility
```

Comment:
>GenericUtility [](http://example.com/codeflow?start=24&length=14)

Why not name this class "IpTablesBase"? Would make more sense, these generic "utility" names can be really confusing